bentoml/deployment/sagemaker/__init__.py

```diff
@@ -206,14 +207,12 @@ def _parse_aws_client_exception_or_raise(e):
     error_log_message = 'AWS ClientError for {operation}: {code} - {message}'.format(
         operation=e.operation_name, code=error_code, message=error_message
     )
+    logger.error(error_log_message)
     if error_code == 'ValidationException':
-        logger.error(error_log_message)
-        return Status.NOT_FOUND(error_response.get('Message', 'Unknown'))
+        return Status.NOT_FOUND(error_log_message)
     elif error_code == 'InvalidSignatureException':
-        logger.error(error_log_message)
-        return Status.UNAUTHENTICATED(error_response.get('Message', 'Unknown'))
+        return Status.UNAUTHENTICATED(error_log_message)
     else:
-        logger.error(error_log_message)
         raise e
```

Comment:
Realized an issue when reading through the tests, here it should raise an Exception that is inherited from BentoMLException, only that way the RPC server can return a response with INTERNAL ERROR code I believe

```
raise BentoMLException(
            'Error interacting with AWS service: {}'.format(str(e))
        )
```